ts/a11y/explorer/Highlighter.ts

```diff
@@ -0,0 +1,485 @@
+    const rect = document.createElementNS('http://www.w3.org/2000/svg', 'rect');
+    rect.setAttribute(
+      'sre-highlighter-added', // Mark highlighting rect.
+      'true'
+    );
+    const padding = 40;
+    const bbox: SVGRect = (node as any as SVGGraphicsElement).getBBox();
+    rect.setAttribute('x', (bbox.x - padding).toString());
+    rect.setAttribute('y', (bbox.y - padding).toString());
+    rect.setAttribute('width', (bbox.width + 2 * padding).toString());
+    rect.setAttribute('height', (bbox.height + 2 * padding).toString());
+    const transform = node.getAttribute('transform');
+    if (transform) {
+      rect.setAttribute('transform', transform);
+    }
+    rect.setAttribute('fill', this.background);
+    node.setAttribute(this.ATTR, 'true');
+    node.parentNode.insertBefore(rect, node);
+    info = { node: node, foreground: node.getAttribute('fill') };
+    if (node.nodeName !== 'rect') {
+      // We currently do not change foreground of collapsed nodes.
+      node.setAttribute('fill', this.foreground);
+    }
+    return info;
+  }
+
+  /**
+   * @override
+   */
+  public setHighlighted(node: HTMLElement) {
+    if (node.tagName === 'svg') {
+      super.setHighlighted(node);
+    }
+  }
+
+  /**
+   * @override
+   */
+  public unhighlightNode(info: Highlight) {
+    const previous = info.node.previousSibling as HTMLElement;
+    if (previous && previous.hasAttribute('sre-highlighter-added')) {
+      info.foreground
+        ? info.node.setAttribute('fill', info.foreground)
+        : info.node.removeAttribute('fill');
+      info.node.parentNode.removeChild(previous);
+      return;
+    }
+    info.node.style.backgroundColor = info.background;
+    info.node.style.color = info.foreground;
+  }
+
+  /**
+   * @override
+   */
+  public isMactionNode(node: HTMLElement) {
+    return node.getAttribute('data-mml-node') === this.mactionName;
+  }
+
+  /**
+   * @override
+   */
+  public getMactionNodes(node: HTMLElement) {
+    return Array.from(
+      node.querySelectorAll(`[data-mml-node="${this.mactionName}"]`)
+    ) as HTMLElement[];
+  }
+}
+
+class ChtmlHighlighter extends AbstractHighlighter {
+  /**
+   * @override
+   */
+  constructor() {
+    super();
+    this.mactionName = 'mjx-maction';
+  }
+
+  /**
+   * @override
+   */
+  public highlightNode(node: HTMLElement) {
+    const info = {
+      node: node,
+      background: node.style.backgroundColor,
+      foreground: node.style.color,
+    };
+    if (!this.isHighlighted(node)) {
+      node.style.backgroundColor = this.background;
+      node.style.color = this.foreground;
+    }
+    return info;
+  }
+
+  /**
+   * @override
+   */
+  public unhighlightNode(info: Highlight) {
+    info.node.style.backgroundColor = info.background;
+    info.node.style.color = info.foreground;
+  }
+
+  /**
+   * @override
+   */
+  public isMactionNode(node: HTMLElement) {
+    return node.tagName?.toUpperCase() === this.mactionName.toUpperCase();
```

Comment:
How about using `toLowerCase()` and not having to uppercase `this.mactionName` every time?